Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ExpressionFilter.cs

```diff
@@ -0,0 +1,628 @@
+            public override int GetHashCode() => _hashCode;
+            public override bool Equals(object obj)
+            {
+                if (!(obj is PredicateValueExpression expression))
+                {
+                    return false;
+                }
+                return _toString.Equals(expression.ToString());
+            }
+        }
+
+        private readonly T _trueValue;
+        private readonly T _falseValue;
+        private readonly string _filterTypeName;
+        private readonly int _hashCode;
+        private string _toString;
+
+        public ExpressionFilter(StringToValueConversion<T> converter, string typeName)
+        {
+            Converter = converter;
+            _filterTypeName = typeName;
+            _hashCode = _filterTypeName.GetHashCode();
+            converter(Tokens.True, out _trueValue, typeName);
+            converter(Tokens.False, out _falseValue, typeName);
+        }
+
+        private HashSet<IRangeClauseExpression> LikePredicates { get; } = new HashSet<IRangeClauseExpression>();
+
+        private HashSet<PredicateValueExpression> ComparablePredicates { get; } = new HashSet<PredicateValueExpression>();
+
+        private bool IsDirty { set; get; } = true;
+
+        protected Dictionary<VariableClauseTypes, HashSet<string>> Variables { get; } = new Dictionary<VariableClauseTypes, HashSet<string>>()
+        {
+            [VariableClauseTypes.Is] = new HashSet<string>(),
+            [VariableClauseTypes.Predicate] = new HashSet<string>(),
+            [VariableClauseTypes.Range] = new HashSet<string>(),
+            [VariableClauseTypes.Value] = new HashSet<string>(),
+        };
+
+        protected StringToValueConversion<T> Converter { set; get; } = null;
+
+        protected HashSet<T> SingleValues { set; get; } = new HashSet<T>();
+
+        protected HashSet<(T Start, T End)> Ranges { set; get; } = new HashSet<(T Start, T End)>();
+
+        protected FilterLimits<T> Limits { get; } = new FilterLimits<T>();
+
+        private Dictionary<string, IExpressionFilter> ComparablePredicateFilters { set; get; } = new Dictionary<string, IExpressionFilter>();
+
+        private Dictionary<string, IExpressionFilter> ComparablePredicateFiltersInverse { set; get; } = new Dictionary<string, IExpressionFilter>();
+
+        public void AddComparablePredicateFilter(string variable, string variableTypeName)
+        {
+            if (variable is null || variable.Length == 0 || variableTypeName is null || variableTypeName.Length == 0)
+            {
+                return;
+            }
+
+            if (!ComparablePredicateFilters.ContainsKey(variable))
+            {
+                ComparablePredicateFilters.Add(variable, ExpressionFilterFactory.Create(variableTypeName));
+                ComparablePredicateFiltersInverse.Add(variable, ExpressionFilterFactory.Create(variableTypeName));
+            }
+        }
+
+        public override bool Equals(object obj)
```

Comment:
Seems redundant; we have a override already defined on the line 47.

Reply:
The overrides on lines 45 - 47 are for the private `PredicateValueExpression` struct used to store info about comparable predicates (e.g., `x < 5`).  The overrides identified here are for `ExpressionFilter<T>`.